src/class-post-republisher.php

```diff
@@ -216,7 +216,11 @@ public function clean_up_after_redirect() {
 
 			\check_admin_referer( 'dp-republish', 'dpnonce' );
 
-			$this->delete_copy( $copy_id, $post_id );
+			if ( \intval( \get_post_meta( $copy_id, '_dp_has_been_republished', true ) ) === 1 ) {
+				$this->delete_copy( $copy_id, $post_id );
+			} else {
```

Comment:
A Yoast best practice is to avoid if/else when possible e,g, by returning within the if. Since Duplicate Post doesn't fully adopt the Yoast CS I guess this can stay for now.